pkg/secretutils/secret_manager.go

```diff
@@ -12,6 +12,13 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	metal3v1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/metal3-io/baremetal-operator/pkg/utils"
+)
+
+const (
+	SecretsFinalizer = metal3v1alpha1.BareMetalHostFinalizer + "/secret"
```

Comment:
I don't think we should add the /secret suffix on here; better to use the finalizer that is already defined in the API module.

Reply:
For core objects like secret (does not apply to objects created via CRD, like BMH):

> Finalizer must be fully qualified since there's a validation process, so it must be declared according to the pattern prometherion/whatever_you_want, taking care the first part must adhere to DNS-1123 specification.